src/context/reducer.ts

```diff
@@ -0,0 +1,17 @@
+import { GlobalState, Types } from "./context.interface";
+
+export const rootReducer = (
+  state: GlobalState,
+  action: { type: Types; payload?: any }
+) => {
+  switch (action.type) {
+    case Types.REQUEST:
+      return { ...state, loading: true };
+    case Types.LOADER_OFF:
+      return { ...state, loading: false };
+    case Types.SET_USER:
+      return { ...state, user: action.payload, loading: false };
```

Comment:
Let's hold off from mutating `state.loading` here

Reply:
when user is set i want to set loader off right dispatching loader function separately will rerender component 1 extra time